localization/tree_structured_parzen_estimator/test/test_tpe.cpp

```diff
@@ -32,34 +32,34 @@ TEST(TreeStructuredParzenEstimatorTest, TPE_is_better_than_random_search_on_sphe
   constexpr int64_t kInnerTrialsNum = 100;
   std::cout << std::fixed;
   std::vector<double> mean_scores;
-  for (const int64_t n_startup_trials : {kInnerTrialsNum, kInnerTrialsNum / 10}) {
-    const std::string method = ((n_startup_trials == kInnerTrialsNum) ? "Random" : "TPE");
+  const int64_t n_startup_trials = kInnerTrialsNum / 10;
+  const std::string method = ((n_startup_trials == kInnerTrialsNum) ? "Random" : "TPE");
 
-    std::vector<double> scores;
-    for (int64_t i = 0; i < kOuterTrialsNum; i++) {
-      double best_score = std::numeric_limits<double>::lowest();
-      const std::vector<bool> is_loop_variable(6, false);
-      TreeStructuredParzenEstimator estimator(
-        TreeStructuredParzenEstimator::Direction::MAXIMIZE, n_startup_trials, is_loop_variable);
-      for (int64_t trial = 0; trial < kInnerTrialsNum; trial++) {
-        const TreeStructuredParzenEstimator::Input input = estimator.get_next_input();
-        const double score = -sphere_function(input);
-        estimator.add_trial({input, score});
-        best_score = std::max(best_score, score);
-      }
-      scores.push_back(best_score);
+  std::vector<double> scores;
+  for (int64_t i = 0; i < kOuterTrialsNum; i++) {
+    double best_score = std::numeric_limits<double>::lowest();
+    const std::vector<double> sample_mean(5, 0.0);
+    const std::vector<double> sample_stddev(5, 2.0);
+    TreeStructuredParzenEstimator estimator(
+      TreeStructuredParzenEstimator::Direction::MAXIMIZE, n_startup_trials, sample_mean,
+      sample_stddev);
+    for (int64_t trial = 0; trial < kInnerTrialsNum; trial++) {
+      const TreeStructuredParzenEstimator::Input input = estimator.get_next_input();
+      const double score = -sphere_function(input);
+      estimator.add_trial({input, score});
+      best_score = std::max(best_score, score);
     }
+    scores.push_back(best_score);
+  }
 
-    const double sum = std::accumulate(scores.begin(), scores.end(), 0.0);
-    const double mean = sum / scores.size();
-    mean_scores.push_back(mean);
-    double sq_sum = 0.0;
-    for (const double score : scores) {
-      sq_sum += (score - mean) * (score - mean);
-    }
-    const double stddev = std::sqrt(sq_sum / scores.size());
-
-    std::cout << method << ", mean = " << mean << ", stddev = " << stddev << std::endl;
+  const double sum = std::accumulate(scores.begin(), scores.end(), 0.0);
+  const double mean = sum / scores.size();
+  mean_scores.push_back(mean);
+  double sq_sum = 0.0;
+  for (const double score : scores) {
+    sq_sum += (score - mean) * (score - mean);
   }
-  ASSERT_LT(mean_scores[0], mean_scores[1]);
+  const double stddev = std::sqrt(sq_sum / scores.size());
```

Comment:
@KYabuuchi 
That's a very valid point.
Since TPE was changed to an NDT-specific implementation, it did not perform well for general functions in the test, so comparisons were removed.
However, after some trial and error, I found that setting the standard deviation somewhat similar to NDT worked well.

Fixed with the following commit : https://github.com/autowarefoundation/autoware.universe/pull/6990/commits/acd71c40d9dca92551f96823f107e8cb37e29ebd

Confirmed that the test succeeded 200 times in a row.

## Experiment method

make `test_loop.sh`

```bash
#!/bin/bash
set -eux

TEST_TARGET_PKG=$1
LOOP_NUM=${2:-100}

colcon build --symlink-install \
  --cmake-args \
    -DCMAKE_BUILD_TYPE=Release \
    -DBUILD_TESTING=ON \
    -DCMAKE_CXX_FLAGS='-fprofile-arcs -ftest-coverage' \
    -DCMAKE_C_FLAGS='-fprofile-arcs -ftest-coverage' \
  --packages-select $TEST_TARGET_PKG

for ((i=1; i<=LOOP_NUM; i++))
do
  colcon test --packages-select ${TEST_TARGET_PKG} --event-handlers console_cohesion+ --return-code-on-test-failure
  echo "Finish test $i"
done
```

and execute it

```bash
~/test_loop.sh tree_structured_parzen_estimator 200
```

Reply:
Great! Now that is an effective test :clap: